internal/util/timerecord/group_checker_test.go

```diff
@@ -37,7 +38,10 @@ func TestGroupChecker(t *testing.T) {
 	})
 	gc2.Check("2")
 
-	assert.Equal(t, 500*time.Millisecond, gc2.d)
+	assert.Equal(t, duration, gc2.d)
+
+	// make sure that the check time of "1" and "2" are all out-of-date.
+	time.Sleep(duration)
```

Comment:
This wouldn't help, if "1" expired but "2" not, and the group checker started to check, it will write ["1"] into signal, and then later write ["1", "2"] after "2" also expired

Reply:
Just rebase the master, 500ms is long enough to catch both "1" and "2"

Reply:
Guess your base doesn't contain the 500ms (which was 10ms) modification

Reply:
Yeah, you are right that this patch won't help, thx for the explanation. So how about change the logic to wati unitl the signal received both "1" and "2"? By the way, I already tested the logic with the latest master branch, it seems that `500ms` didn't help too.